tests/Adapter/UpdaterTestCase.php

```diff
@@ -925,6 +925,7 @@ public function crawl()
                 <<<'EOT'
 class Aardvark
 {
+
     /**
```

Comment:
hmm, actually this change is controversial for me and is not the SF coding standard or PSR-12, would rather have no space here.

Reply:
I would love to too :)
The thing is that I trade a "bug" for another one.
Before the changes if you had
```php
<?php

class Foo
{
    use SomeTrait;

    public function __construct($a)
    {
    }
}
```

And you try to complete the constructor, the property would have been added above the trait declaration, which is not PSR-12 compliant neither.

The problem here is that we want to add a blank line only if there is a phpdoc block for the new property.
But the phpdoc part is totally up to the template, so the updater can't know when to add one or not.
And the templates can't know what is the preview member to decide if the blank line should be avoid or not.

Reply:
hmm, only thing I can suggest is making it as consistent as possible, if we trade one bug for another then let's not add this change.

The problem in general with this library is that it explicitly couples to the presenatation - one solution might be to write a simple CS/whitespace fixer using the Tolerant Parser AST, which can operate on a range of lines.

We could then care less about whitespace concerns when building the source code.

Reply:
I think it's consistent in the way that it's only in two scenario and only if using phpdoc.
Plus it's 100% compliant and bug free for PHP 7.4.
So I would said that the trade is favorable, otherwise we go back with the previous problems in all versions.

And most important it only appends (if I remember correctly), when you have no properties yet.
So it's in a pretty specific and rare case.

The use of the parser would make it clean of course, if we can manage to check the number of black lines between a declaration member and a function member.
I'm not sure how though, will have to check.

Reply:
hmm, ok. if the pros outweigh the cons, the let's go with it.

> The user of the parser would ...

I'll try and have a look at this